File: foliage-sample/src/main/java/org/lavenderx/foliage/sample/listener/HelloRpcRpcMessageListener.java
package org.lavenderx.foliage.sample.listener;

import lombok.extern.slf4j.Slf4j;
import org.lavenderx.foliage.nettyrpc.annotation.RpcListenerContainer;
import org.lavenderx.foliage.nettyrpc.rpcservice.HelloRpcService;
import org.springframework.stereotype.Service;

@Service
@Slf4j
@RpcListenerContainer(value = HelloRpcService.class)
public class HelloRpcRpcMessageListener implements HelloRpcService {

    @Override
    public String hello(String name) {
        return "Hello, " + name + "!";
    }
}

File: foliage-sample/src/main/kotlin/org/lavenderx/foliage/sample/listener/HelloRpcRpcMessageListener.kt
package org.lavenderx.foliage.sample.listener

import io.foliage.netty.rpc.annotation.RpcListenerContainer
import io.foliage.netty.rpc.rpcservice.HelloRpcService
import org.springframework.stereotype.Service

@Service
@RpcListenerContainer(HelloRpcService::class)
class HelloRpcRpcMessageListener : HelloRpcService {

    override fun hello(name: String): String {
        return "Hello, $name!"
    }
}